cardano-node/app/cardano-node.hs

```diff
@@ -152,33 +144,54 @@ nodeCliParser = do
   reqNetMagic <- parseRequireNetworkMagic
   slotLength <- parseSlotLength
 
-  pure $ NodeCLI topInfo nAddr ptcl vMode logCliArgs traceOptions
-         (createPcc dbPath socketDir genPath genHash delCert
-                   sKey pbftSigThresh reqNetMagic slotLength)
+  pure $ NodeCLI
+         (createPcc dbPath socketDir topInfo nAddr ptcl logConfigFp vMode
+                    logMetrics genPath genHash delCert sKey pbftSigThresh
+                    reqNetMagic traceOptions slotLength)
  where
   -- This merges the command line parsed values into one `PartialCardanoconfiguration`.
   createPcc
     :: Last FilePath
     -> Last FilePath
+    -> Last TopologyInfo
+    -> Last NodeAddress
+    -> Last Protocol
+    -> Last FilePath
+    -> Last ViewMode
+    -> Last Bool
     -> Last FilePath
     -> Last Text
     -> Last FilePath
     -> Last FilePath
     -> Last Double
     -> Last RequireNetworkMagic
+    -> Last TraceOptions
     -> Last Consensus.SlotLength
     -> PartialCardanoConfiguration
   createPcc
     dbPath
     socketDir
+    topInfo
+    nAddr
+    ptcl
+    logConfigFp
+    vMode
+    logMetrics
     genPath
     genHash
     delCert
     sKey
     pbftSigThresh
     reqNetMagic
+    traceOptions
     slotLength = mempty { pccDBPath = dbPath
+                        , pccNodeAddress = nAddr
                         , pccSocketDir = socketDir
+                        , pccTopologyInfo = topInfo
+                        , pccProtocol = ptcl
+                        , pccViewMode = vMode
+                        , pccLogConfig = logConfigFp
+                        , pccLogMetrics = logMetrics
                         , pccCore = mempty { pcoGenesisFile = genPath
```

Comment:
This looked kind of strange the first time I saw it, but now (since you kept adding things to it) looks even stranger.
Doesn't `mergeConfiguration` does most of these things?